src/autopas/tuning/tuningStrategy/fuzzyTuning/fuzzyController/LinguisticVariable.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file LinguisticVariable.h
+ * @author Manuel Lerchner
+ * @date 18.04.24
+ */
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "CrispSet.h"
+#include "FuzzySet.h"
+
+namespace autopas::fuzzy_logic {
+
+/**
+ * A class representing a LinguisticVariable. A LinguisticVariable is defined on a CrispSet and consists of several
+ * FuzzySets, which are the linguistic terms of the LinguisticVariable.
+ */
+class LinguisticVariable {
+ public:
+  /**
+   * Constructs a LinguisticVariable with the given name and range.
+   * @param name The name of the LinguisticVariable.
+   * @param range The range of the LinguisticVariable in the form [min, max].
+   */
+  explicit LinguisticVariable(const std::string &name, std::pair<double, double> range);
+
+  /**
+   * Adds a new linguistic term to the LinguisticVariable.
+   * Additionally updates the CrispSet of the linguistic term to the current CrispSet.
+   * @param linguisticTerm The linguistic term to add.
+   */
+  void addLinguisticTerm(const std::shared_ptr<FuzzySet> &linguisticTerm);
+
+  /**
+   * Overload of the operator== to extract a linguistic term by its name. This allows a very concise syntax to create
+   * fuzzy rules.
+   * @param linguisticTerm The name of the linguistic term to extract.
+   * @return The FuzzySet with the given name.
+   */
+  std::shared_ptr<FuzzySet> operator==(const std::string &linguisticTerm) const;
```

Comment:
It is non standard for C++ but it tries to mimic the mathematical way of writing this. (Same with `operator&&`, `operator||` and `operator!` btw.)

On paper you would also write $IF\ age=low\ THEN\ height=small$ even if the equality doesn't make sense (as you said: `age/height` are linguistic variables, `low/small` are linguistic terms). The $=$ operator just got overloaded to write the rules more human-like.

When overloading the `operator==` in C++ we can also use this direct, mathematical syntax in the code and write the rules as:

```cpp
auto X1 = LinguisticVariable("x1", std::pair(-5, 30));
auto X2 = LinguisticVariable("x2", std::pair(-5, 30));
auto Y = LinguisticVariable("y", std::pair(-5, 30));

auto x1Low = FuzzySetFactory::makeFuzzySet("low", "Triangle", {0, 10, 20});
auto x1High = FuzzySetFactory::makeFuzzySet("high", "Triangle", {15, 25, 30});
auto x2Low = FuzzySetFactory::makeFuzzySet("low", "Triangle", {0, 10, 20});
auto x2High = FuzzySetFactory::makeFuzzySet("high", "Triangle", {15, 25, 30});
auto yLow = FuzzySetFactory::makeFuzzySet("low", "Triangle", {0, 10, 20});
auto yHigh = FuzzySetFactory::makeFuzzySet("high", "Triangle", {15, 25, 30});

X1.addLinguisticTerm(x1Low);
X1.addLinguisticTerm(x1High);
X2.addLinguisticTerm(x2Low);
X2.addLinguisticTerm(x2High);
Y.addLinguisticTerm(yLow);
Y.addLinguisticTerm(yHigh);

//overloading the operator== and operator&& allows this syntax. 
auto rule1 = FuzzyRule(X1 == "low" && X2 == "high", Y == "low");
```

The `extract` descriptions comes from the fact that the `operator==` returns the **FuzzySet** corresponding to the linguistic term you provide. 
In a way a linguistic variable, is just a container for the different FuzzySets and the `operator==` selects the one corresponding to the name you provide.

I wouldn't change this as it would break all the nice syntax used for writing rules.

Reply:
Okay, looks good. But I made a documentation suggestion above